analysis_driver/segmentation.py

```diff
@@ -17,31 +18,25 @@ def serialize(self, x):
 
 class BasicStage(luigi.Task, AppLogger):
     __stagename__ = None
-    previous_stages = []
     exit_status = None
 
+    previous_stages = EGCGParameter(default=[])
```

Comment:
Instead of defining `previous_stages` in the class declaration for  the stage, we can now create stage objects directly and pass them into each other as parameters.